thunder/core/proxies.py

```diff
@@ -1251,7 +1256,7 @@ def _infer_tensor_properties(
     else:
         # deferred computation of numel
         # TODO: similar to how `shape` is handled, this should be CSE or lifted for efficiency
-        _numel = lambda tp: reduce(operator.mul, tp.shape, 1)
+        _numel = lambda *args: reduce(operator.mul, _shape, 1)
 
     # TODO Alias rank to ndim?
     _ndim = len(_shape)
```

Comment:
hrm, unrelated to this commit, but to rehash the previous point: when `_shape` is `None` this is going to break.